src/transformers/models/marian/modeling_marian.py

```diff
@@ -1132,6 +1132,13 @@ def forward(
 
             >>> last_hidden_states = outputs.last_hidden_state
         """
+        # different to other models, Marian automatically creates decoder_input_ids from
```

Comment:
This shouldn't be added here. This is done for BART because of the denoising pre-training objective, so only BART and BART prepare the `deocder_input_ids` from `input_ids`.  For Marian users should pass either `decoder_input_ids` or `labels` and this is already handled by the `MarianMTModel` class.

Reply:
> This shouldn't be added here. This is done for BART because of the denoising pre-training objective, so only BART and mBART prepare the `deocder_input_ids` from `input_ids`. For Marian users should pass either `decoder_input_ids` or `labels` and this is already handled by the `MarianMTModel` class.

Thank you @patil-suraj for the reply, but as I said in [this](https://github.com/huggingface/transformers/issues/13823#issuecomment-932349647) comment, `transformers.onnx.convert.export()` calls `MarianMTModel.forward()` without passing `decoder_input_ids` or `labels`, so how is this supposed to be handled?